packages/react/src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -652,6 +685,10 @@ const MultiSelect = React.forwardRef(
       selectedItems.length > 0 &&
       selectedItems.map((item) => (item as selectedItemType)?.text);
 
+    const selectedItemsWithoutSelectAll = selectedItems.filter(
+      (item: any) => !item.isSelectAll
+    );
```

Comment:
Should this be guarded behind `selectAll`? If so, the usages will need to be updated.

Reply:
I believe we do not need a guard for this as it's a simple filtering which will remove items with isSelectAll key-value, which will pass anyway if user would not pass an object with isSelectAll